selectors/global_header.ts

```diff
@@ -5,12 +5,14 @@ import Constants from 'utils/constants';
 import {get} from 'mattermost-redux/selectors/entities/preferences';
 import {GlobalState} from 'types/store';
 import {getFeatureFlagValue} from 'mattermost-redux/selectors/entities/general';
+import {isMobileViewPortSize, isMobile} from '../utils/user_agent';
 
 const Preferences = Constants.Preferences;
 
 export function getGlobalHeaderEnabled(state: GlobalState): boolean {
     const featureFlagEnabled = getFeatureFlagValue(state, 'GlobalHeader') === 'true';
     const userPreferenceEnabled = get(state, Preferences.CATEGORY_DISPLAY_SETTINGS, Preferences.GLOBAL_HEADER_DISPLAY, Preferences.GLOBAL_HEADER_DISPLAY_DEFAULT) === Preferences.GLOBAL_HEADER_DISPLAY_ON;
+    const isMobileView = isMobileViewPortSize() || isMobile();
 
-    return featureFlagEnabled && userPreferenceEnabled;
+    return !isMobileView && featureFlagEnabled && userPreferenceEnabled;
```

Comment:
In a vacuum this won't work in the case of a viewport size change, since I don't think this will trigger a re-render on its own. But there is probably enough going on and re-rendering taking place that this will work anyway. Not sure we even really care about the viewport size change case.

Reply:
This is a temporary change anyways and it is working for now. I agree that this is not the best option, but just hiding the GH is not gonna work, since the elements were re-arranged quite a bit and this prevents that from showing.

Reply:
One of these days, we should move whether or not the app is in mobile view into redux so that we can easily access them in places like this. We did something like that in the mobile app ages ago, and it made it a lot easier to reference the screen size.

Reply:
@hmhealey agreed to 100%!